### PR TITLE
Add rz_analysis_function_set_cc to set the calling convention

### LINKED ISSUE #5672: Add an API to change the function convention
Create an API to modify `RzAnalysisFunction->cc` nicely

Necessary for providing this feature in Cutter: https://github.com/rizinorg/cutter/issues/3513

### PR DESCRIPTION
**Pull request by Ayushd785**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/dev/CONTRIBUTING.md) to this repository.
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#code-style).
- [x] I've documented every `RZ_API` function and struct this PR changes.
- [x] I've added tests that prove my changes are effective (required for changes to `RZ_API`).
- [ ] I've updated the [Rizin book](https://github.com/rizinorg/book) with the relevant information (if needed).
- [x] I've used AI tools to generate fully or partially these code changes and I'm sure the changes are not copyrighted by somebody else. 

**Detailed description**
- Added `rz_analysis_function_set_cc()` API function in `librz/arch/fcn.c`
  - Validates that the calling convention exists before setting
  - Uses constpool for memory efficiency (consistent with existing code)
  - Handles NULL/empty strings to clear the calling convention
  - Returns `true` on success, `false` on failure with appropriate error logging
- Updated the command handler in `librz/core/cmd/cmd_analysis.c` to use the new API instead of direct field assignment
- Added comprehensive test suite covering all edge cases

**Closing issues**
closes #5672

### CODE REVIEW

**Review by notxvilka — APPROVED** [review]

File: librz/arch/fcn.c [hunk 1]
```diff
@@ -2040,6 +2040,33 @@ RZ_API bool rz_analysis_function_set_type_str(RzAnalysis *a, RZ_NONNULL RzAnalys
...
+ *
+ * \param analysis RzAnalysis instance
+ * \param fcn Function to update
+ * \param cc Calling convention name, or NULL to clear
+ * \return true on success, false if the calling convention doesn't exist
+ */
+RZ_API bool rz_analysis_function_set_cc(RzAnalysis *analysis, RZ_NONNULL RzAnalysisFunction *fcn, RZ_NULLABLE const char *cc) {
+	rz_return_val_if_fail(analysis && fcn, false);
+
+	if (!cc || !*cc) {
```
Comment (wargio) [inline librz/arch/fcn.c]: RZ_STR_EMPTY(cc)

**Review by wargio — CHANGES_REQUESTED** [review]
small change needed, but overall mergiable

**Comment by Ayushd785** [comment]
> small change needed, but overall mergiable

Okay i'll fix it

**Comment by Ayushd785** [comment]
> small change needed, but overall mergiable

I have implemented your suggestion. Thank you for the review!

**Review by wargio — APPROVED** [review]
